### PR TITLE
Feat/refactor server

### PR DESCRIPTION
**Pull request by alexanderattar**
This PR is a refactor of the server file into a `Server` class to encapsulate and simplify the logic

### CODE REVIEW

File: test/test-utils.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ function bn (number) {
 
 describe('utils', function () {
   describe('addRange', () => {
-    it('adds ranges and merges if possible', async () => {
+    it('adds ranges and merges if possible', async (done) => {
```
Comment (kfichter) [inline test/test-utils.js]: Hmm is this necessary?

Comment (alexanderattar) [reply test/test-utils.js]: Hmm maybe not actually. There is an issue with the tests hanging that I was trying to fix, but this didn't seem to do the trick. I believe it is this test that is causing the hanging though. I can revert it for now

Comment (alexanderattar) [reply test/test-utils.js]: I went ahead and reverted these changes

**Comment by alexanderattar** [comment]
I went ahead and fixed the merge conflicts also so this should be good to merge

**Comment by karlfloersch** [comment]
Really appreciate the refactor thank you alex!!! <3

**Comment by alexanderattar** [comment]
Dope! Happy to help. It was fun times.



*Alexander Attar* // Platform Lead // Ujo
alexander.attar@consensys.net <alexander.attar@consenSys.net>
49 Bogart St, Suite 22, Brooklyn NY 11206
Web <https://consensys.net/> | Twitter <https://twitter.com/ConsenSysLLC> |
Facebook <https://www.facebook.com/consensussystems> | Linkedin
<https://www.linkedin.com/company/consensus-systems-consensys-> | Newsletter
<http://consensys.us11.list-manage.com/subscribe?u=947c9b18fc27e0b00fc2ad055&id=257df01285&utm_content=buffer1ce12&utm_medium=social&utm_source=facebook.com&utm_campaign=buffer>

**Pull request merged**